community/io/src/test/java/org/neo4j/adversaries/pagecache/AdversarialReadPageCursor.java

```diff
@@ -286,14 +413,92 @@ public boolean checkAndClearBoundsFlag()
     }
 
     @Override
+    public void checkAndClearCursorError() throws CursorException
+    {
+        delegate.checkAndClearCursorError();
+    }
+
+    @Override
     public void raiseOutOfBounds()
     {
         delegate.raiseOutOfBounds();
     }
 
     @Override
+    public void setCursorError( String message )
+    {
+        delegate.setCursorError( message );
+    }
+
+    @Override
+    public void clearCursorError()
+    {
+        delegate.clearCursorError();
+    }
+
+    @Override
     public PageCursor openLinkedCursor( long pageId )
     {
-        return linkedCursor = new AdversarialReadPageCursor( delegate.openLinkedCursor( pageId ), adversary );
+        return linkedCursor = new AdversarialReadPageCursor( delegate.openLinkedCursor( pageId ), state );
+    }
+
+    @Override
+    public String toString()
+    {
+        State s = this.state;
+        StringBuilder sb = new StringBuilder();
+        for ( Object o : s.inconsistentReadHistory )
+        {
+            sb.append( o.toString() ).append( '\n' );
+            if ( o instanceof NumberValue )
```

Comment:
why do we need to do cast here at all and invoke type specific `printStackTrace` method instead of just do it as part of NumberValue toString call?